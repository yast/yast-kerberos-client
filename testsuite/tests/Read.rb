# encoding: utf-8

#  Read.ycp
#  Test of Kerberos::Read function (whole read process, many config files)
#  Author:	Jiri Suchomel <jsuchome@suse.cz>
#  $Id$
module Yast
  class ReadClient < Client
    def main
      # testedfiles: Kerberos.ycp Pam.ycp
      Yast.import "Testsuite"
      Yast.import "Kerberos"

      @READ = {
        "etc"       => {
          # /etc/krb5.conf
          "krb5_conf" => {
            "v" => {
              "libdefaults" => {
                "default_realm" => ["SUSE.CZ"],
                "clockskew"     => ["500"]
              },
              "SUSE.CZ"     => {
                "kdc"            => ["chimera.suse.cz", "kdc.suse.cz"],
                "admin_server"   => nil,
                "default_domain" => nil
              },
              "pam"         => {
                "minimum_uid"       => ["1"],
                "renew_lifetime"    => nil,
                "ticket_lifetime"   => nil,
                "forwardable"       => nil,
                "proxiable"         => nil,
                "use_authtok"       => nil,
                "keytab"            => nil,
                "ccache_dir"        => nil,
                "ccname_template"   => nil,
                "mappings"          => nil,
                "existing_ticket"   => ["true"],
                "external"          => ["false"],
                "validate"          => nil,
                "use_shmem"         => nil,
                "addressless"       => nil,
                "debug"             => nil,
                "debug_sensitive"   => nil,
                "initial_prompt"    => nil,
                "subsequent_prompt" => nil,
                "addressless"       => ["false"],
                "banner"            => ["haha"]
              },
              "pkinit"      => { "trusted_servers" => nil }
            }
          },
          # /etc/security/pam_*
          "security"  => {
            "section" => {
              "/etc/security/pam_unix2.conf"   => {},
              "/etc/security/pam_pwcheck.conf" => {}
            },
            "v"       => {
              "/etc/security/pam_unix2.conf" => {
                "passwd" => "nullok use_ldap",
                "auth"   => "nullok use_krb5"
              }
            }
          },
          # /etc/ssh/ssh_config
          "ssh"       => {
            "ssh_config" => {
              "s" => { "*" => {} },
              "v" => {
                "*" => {
                  "GSSAPIAuthentication"      => "yes",
                  "GSSAPIDelegateCredentials" => "yes"
                }
              }
            }
          },
          "sssd_conf" => {
            "v" => {
              "domain/default" => {
                "krb5_realm"              => "SUSE.DE",
                "krb5_server"             => nil,
                "krb5_ccachedir"          => nil,
                "krb5_ccname_template"    => nil,
                "krb5_keytab"             => nil,
                "krb5_renewable_lifetime" => nil,
                "krb5_lifetime"           => nil,
                "krb5_validate"           => nil
              }
            }
          }
        },
        "sysconfig" => { "openafs-client" => nil },
        "target"    => { "stat" => { 1 => 2 }, "string" => "" }
      }

      @EX = {
        "target" => {
          "bash"        => 0,
          "bash_output" => {
            # call of ypdomainname
            #		"stdout" : "suse.cz"
            # call of pam-config
            "stdout" => "password: "
          }
        }
      }

      Testsuite.Dump("==== reading... ============================")

      Testsuite.Test(lambda { Kerberos.Read }, [@READ, {}, @EX], 0)

      Testsuite.Dump("============================================")

      Testsuite.Dump(
        Builtins.sformat("kerberos used: %1", Kerberos.use_pam_krb)
      )

      Testsuite.Dump(
        Builtins.sformat("default realm: %1", Kerberos.default_realm)
      )
      Testsuite.Dump(Builtins.sformat("kdc: %1", Kerberos.kdc))

      Testsuite.Dump(
        Builtins.sformat("\nssh support: %1", Kerberos.ssh_support)
      )
      Testsuite.Dump("============================================")

      #    READ ["etc", "krb5_conf", "v", "libdefaults", "default_realm"]	= nil;
      Ops.set(@READ, ["target", "stat"], {})

      Testsuite.Dump("==== reading... ============================")

      Testsuite.Test(lambda { Kerberos.Read }, [@READ, {}, @EX], 0)

      Testsuite.Dump(
        Builtins.sformat("default realm: %1", Kerberos.default_realm)
      )

      Testsuite.Dump("============================================")

      nil
    end
  end
end

Yast::ReadClient.new.main
