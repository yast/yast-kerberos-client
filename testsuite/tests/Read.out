Dump	==== reading... ============================
Execute	.target.bash_output "/usr/sbin/pam-config -q --krb5" $["stdout":"password: "]
Dir	.etc.krb5_conf.s: []
Read	.etc.krb5_conf.v.libdefaults.default_realm ["SUSE.CZ"]
Read	.etc.krb5_conf.v.libdefaults.clockskew ["500"]
Read	.etc.krb5_conf.v."SUSE.CZ"."kdc" ["chimera.suse.cz", "kdc.suse.cz"]
Read	.etc.krb5_conf.v."SUSE.CZ"."admin_server" nil
Read	.etc.krb5_conf.v."SUSE.CZ"."default_domain" nil
Read	.etc.krb5_conf.v.pam."ticket_lifetime" nil
Read	.etc.krb5_conf.v.pam."renew_lifetime" nil
Read	.etc.krb5_conf.v.pam."forwardable" nil
Read	.etc.krb5_conf.v.pam."proxiable" nil
Read	.etc.krb5_conf.v.pam."minimum_uid" ["1"]
Read	.etc.krb5_conf.v.pam."keytab" nil
Read	.etc.krb5_conf.v.pam."ccache_dir" nil
Read	.etc.krb5_conf.v.pam."ccname_template" nil
Read	.etc.krb5_conf.v.pam."mappings" nil
Read	.etc.krb5_conf.v.pam."existing_ticket" ["true"]
Read	.etc.krb5_conf.v.pam."external" ["false"]
Read	.etc.krb5_conf.v.pam."validate" nil
Read	.etc.krb5_conf.v.pam."use_shmem" nil
Read	.etc.krb5_conf.v.pam."addressless" ["false"]
Read	.etc.krb5_conf.v.pam."debug" nil
Read	.etc.krb5_conf.v.pam."debug_sensitive" nil
Read	.etc.krb5_conf.v.pam."initial_prompt" nil
Read	.etc.krb5_conf.v.pam."subsequent_prompt" nil
Read	.etc.krb5_conf.v.pam."banner" ["haha"]
Read	.etc.krb5_conf.v.pkinit.trusted_servers nil
Dir	.etc.ssh.ssh_config.s: ["*"]
Dir	.etc.ssh.ssh_config.s: ["*"]
Dir	.etc.ssh.ssh_config.v."*": ["GSSAPIAuthentication", "GSSAPIDelegateCredentials"]
Read	.etc.ssh.ssh_config.v."*"."GSSAPIAuthentication" "yes"
Read	.etc.ssh.ssh_config.v."*"."GSSAPIDelegateCredentials" "yes"
Execute	.target.bash_output "/usr/sbin/pam-config -q --sss" $["stdout":"password: "]
Read	.etc.sssd_conf.v."domain/default"."krb5_realm" "SUSE.DE"
Read	.etc.sssd_conf.v."domain/default"."krb5_server" nil
Read	.etc.sssd_conf.v."domain/default"."krb5_ccachedir" nil
Read	.etc.sssd_conf.v."domain/default"."krb5_ccname_template" nil
Read	.etc.sssd_conf.v."domain/default"."krb5_keytab" nil
Read	.etc.sssd_conf.v."domain/default"."krb5_renewable_lifetime" nil
Read	.etc.sssd_conf.v."domain/default"."krb5_lifetime" nil
Read	.etc.sssd_conf.v."domain/default"."krb5_validate" nil
Return	true
Dump	============================================
Dump	kerberos used: true
Dump	default realm: SUSE.DE
Dump	kdc: chimera.suse.cz kdc.suse.cz
Dump	
Dump	ssh support: true
Dump	============================================
Dump	==== reading... ============================
Execute	.target.bash_output "/usr/sbin/pam-config -q --krb5" $["stdout":"password: "]
Execute	.target.bash "/usr/bin/touch /etc/krb5.conf" 0
Dir	.etc.ssh.ssh_config.s: ["*"]
Dir	.etc.ssh.ssh_config.s: ["*"]
Dir	.etc.ssh.ssh_config.v."*": ["GSSAPIAuthentication", "GSSAPIDelegateCredentials"]
Read	.etc.ssh.ssh_config.v."*"."GSSAPIAuthentication" "yes"
Read	.etc.ssh.ssh_config.v."*"."GSSAPIDelegateCredentials" "yes"
Execute	.target.bash_output "/usr/sbin/pam-config -q --sss" $["stdout":"password: "]
Return	true
Dump	default realm: SUSE.DE
Dump	============================================
